src/lib/analysers/meta/main.py

```diff
@@ -0,0 +1,71 @@
+from lib.common.analyser import Analyser
+from lib.common.analyser import paths_to_components
+from lib.common.get_module import get_module
+import os
+from shutil import copyfile, rmtree
+
+
+class MetaAnalyser(Analyser):
+    def setup_run(self, config):
+
+        self.child_analysers = []
+        whitelist = self.CONFIG["elements_in"]
+
+        cmps = paths_to_components(whitelist)
+        self.selector = cmps[0][0]
+
+        child_modules = config["modules"]
+        for module in child_modules:
+            child_name = module["name"]
+            child_config = module["config"]
+            ChildAnalyser = get_module("analyser", child_name)
+            if ChildAnalyser is None:
+                raise Exception(
+                    f"The module you have specified, {child_name}, does not exist"
+                )
+            child_analyser = ChildAnalyser(child_config, child_name, self.FOLDER)
+            child_analyser.setup_run(child_config)
+            self.child_analysers.append(child_analyser)
+            self.logger(f"Setup child analyser {child_name}")
+
+    def analyse_element(self, element, config):
+        src = None
+        child_element = None
+        for index, analyser in enumerate(self.child_analysers):
+            child_element = self._derive_child_element(index, element, src, analyser)
+            analyser.analyse_element(child_element, analyser.CONFIG)
+            el_id = element["id"]
+            self.logger(f"Analysed element {el_id} in {analyser.NAME}")
+            src = child_element["dest"]
+        self._finalise_element(config, child_element, element)
+
+    def post_analyse(self, config):
+        delete_cache = config["delete_cache"]
+        if delete_cache:
+            self.logger("deleting cache")
+            derived_folder = self.get_derived_folder(self.selector)
+            cache = f"{derived_folder}/cache"
+            rmtree(cache)
+
+    def _derive_child_element(self, child_index, element, child_src, analyser):
+        derived_folder = self.get_derived_folder(self.selector)
```

Comment:
I think we actually want to remove `get_derived_folder` from `Analyser` to make the code more functional. It probably should have been taken out in #44. Can you refactor to work this out from the arguments of `analyse_element`?

Reply:
And also remove `get_derived_folder` from `Analyser` if possible.

Reply:
I have refactored to pass the relevant derived_folder data in the `analyse_element` arguments (also in `post_analyse`).

Removing it from Analyser is more problematic, as Analysers are designed to work on (potentially) multiple selectors and so might have multiple derived_folders which need to be computed for each element. For the moment I changed `get_derived_folder` to` __get_derived_folder` to make it privateish.